File: frontend/dashboard/types/workflow.ts
interface WorkflowResponse {
  id: number;
  name: string;
  type: "image_to_image" | "text_to_image" | "comment_to_image";
  json: string;
  image_result_node: string;
  enabled: boolean;
  create_time: string;
}

interface CreateWorkflowRequest {
  name: string;
  type: string;
  json: string;
  image_result_node: string;
}

interface UpdateWorkflowRequest {
  name?: string;
  type?: string;
  json?: string;
  image_result_node?: string;
  enabled?: boolean;
}

File: dto/workflow.go
package dto

type WorkflowCreateRequest struct {
	Name            string `json:"name"`
	Type            string `json:"type"`
	JSON            string `json:"json"`
	ImageResultNode string `json:"image_result_node"`
}

type WorkflowUpdateRequest struct {
	Name            *string `json:"name,omitempty"`
	Type            *string `json:"type,omitempty"`
	JSON            *string `json:"json,omitempty"`
	ImageResultNode *string `json:"image_result_node,omitempty"`
	Enabled         *bool   `json:"enabled,omitempty"`
}

type WorkflowResponse struct {
	ID              int    `json:"id"`
	Name            string `json:"name"`
	Type            string `json:"type"`
	JSON            string `json:"json"`
	ImageResultNode string `json:"image_result_node"`
	Enabled         bool   `json:"enabled"`
	CreateTime      string `json:"create_time"`
}
